app/controllers/settings/design.js

```diff
@@ -40,25 +42,31 @@ export default Controller.extend({
             this.save.perform();
         },
 
-        addNavItem() {
-            let newNavItem = this.newNavItem;
+        addNavItem(secondary) {
+            let itemToAdd = this.newNavItem;
+
+            if (secondary) {
+                itemToAdd = this.newSecondaryNavItem;
+            } else {
+                itemToAdd = this.newNavItem;
+            }
 
             // If the url sent through is blank (user never edited the url)
-            if (newNavItem.get('url') === '') {
-                newNavItem.set('url', '/');
+            if (itemToAdd.get('url') === '') {
+                itemToAdd.set('url', '/');
             }
 
-            return newNavItem.validate().then(() => {
-                this.addNewNavItem();
+            return itemToAdd.validate().then(() => {
+                this.addNewNavItem(secondary);
             });
         },
 
-        deleteNavItem(item) {
+        deleteNavItem(secondary, item) {
```

Comment:
`secondary` isn't strictly necessary here because the information should already be present on `item.isSecondary`

Reply:
I tried that and the item didn't seem to be the model - maybe I missed something

Reply:
Ah - I remember now.

`isSecondary` is only set on the "new" nav items. `isSecondary` is not actually used at all, I thought I'd removed it.

This was the approach that I thought was more correct, but I couldn't figure out how to make it so that the models automatically had isSecondary applied (during the transform?), when they were pulled from the API. 

So what happened was, this approach worked fine for the new items, but when deleting an existing item that wasn't created during the current session, it didn't work because isSecondary was always false.